inflate.c

```diff
@@ -1149,17 +1150,33 @@ int ZEXPORT inflate(z_stream *strm, int flush) {
                 }
                 if (copy > state->length)
                     copy = state->length;
+                if (copy > left)
+                    copy = left;
+                left -= copy;
+                state->length -= copy;
+                if (copy >= sizeof(long)) {
+                    put = chunk_memcpy(put, from, copy);
+                } else {
+                    do {
+                        *put++ = *from++;
+                    } while (--copy);
+                }
             } else {                             /* copy from output */
-                from = put - state->offset;
+                unsigned offset = state->offset;
+                from = put - offset;
                 copy = state->length;
+                if (copy > left)
+                    copy = left;
+                left -= copy;
+                state->length -= copy;
+                if (copy >= sizeof(long)) {
+                    put = chunk_memset(put, from, offset, copy);
```

Comment:
Oh, except this happens right after `left` is modified.  I mean to test the _old_ value of `left`.